### PR TITLE
Update the documentation to mention `deva`

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

During the Hackathon, I introduced a new tool called [`deva`](https://datadoghq.atlassian.net/wiki/spaces/~61a7d8f7ebce470067858b06/pages/3732504868/Datadog+Agent+-+Run+your+invoke+tasks+with+deva) that streamlines the setup process to run invoke tasks locally. 

This PR adds this tool to the official documentation.

### Motivation

We think using it is easier and less invasive than the previous recommended approach.

### Additional Notes

https://datadoghq.atlassian.net/browse/ADXT-297

### CODE REVIEW

File: docs/public/setup.md [hunk 1]
```diff
@@ -33,10 +33,41 @@ On Windows, install Python 2.7 and/or 3.11 via the [official installer](https://
 
 ##### Preface
 
+[Invoke](http://www.pyinvoke.org) is a task runner written in Python that is extensively used in this project to orchestrate builds and test runs. To run the tasks, you need to have it installed on your machine. We offer two different ways to run our invoke tasks.
+
+##### Installation with the `deva` CLI tool (recommended)
```
Comment (ofek) [inline docs/public/setup.md:38]: I would try to shorten headers:

```suggestion
##### `deva` (recommended)
```

File: docs/dev/agent_dev_env.md [hunk 2]
```diff
@@ -7,82 +7,105 @@ To build the agent on Windows, see [datadog-agent-buildimages](https://github.co
...
 #### Python Dependencies
 
 ##### Preface
 
-To protect and isolate your system-wide python installation, a python virtual
-environment is _highly_ recommended (though optional). It will help keep a
-self-contained development environment and ensure a clean system Python.
+[Invoke](http://www.pyinvoke.org) is a task runner written in Python that is extensively used in this project to orchestrate builds and test runs. To run the tasks, you need to have it installed on your machine. We offer two different ways to run our invoke tasks.
+
+##### Installation with the `deva` CLI tool (recommended)
```
Comment (FlorentClarret) [inline docs/dev/agent_dev_env.md:42]:
```suggestion
##### `deva` (recommended)
```

File: docs/dev/agent_dev_env.md [hunk 3]
```diff
@@ -7,82 +7,105 @@ To build the agent on Windows, see [datadog-agent-buildimages](https://github.co
...
+2. Make it executable,
+3. Run the invoke command you need.
+
+The Python environment will automatically be created on the first run. and will be reused for subsequent runs. For example:
+
+```shell
+$ cd datadog-agent
+$ curl -L -o deva https://github.com/DataDog/datadog-agent-devtools/releases/download/deva-v1.0.0/deva-aarch64-unknown-linux-gnu-1.0.0
+$ chmod +x deva
+$ ./deva linter.go
```
Comment (chouetz) [inline docs/dev/agent_dev_env.md:58]: you simply replaced `inv` by `deva` everywhere. Should it be `./deva`? Should you add a note about adding it in the PATH?

File: docs/dev/linters.md [hunk 4]
```diff
@@ -45,7 +45,7 @@ A: This could have several causes:
     - `go version` should output the same as [the repository .go-version](https://github.com/DataDog/datadog-agent/blob/dffd3262934a5540b9bf8e4bd3a743732637ef37/.go-version).
     - `golangci-lint --version` should output the same as [the repository internal/tools/go.mod](https://github.com/DataDog/datadog-agent/blob/dffd3262934a5540b9bf8e4bd3a743732637ef37/internal/tools/go.mod/#L8).
 - You're testing OS specific code; running locally, the linters only get the results for your local OS.
-- You didn't run `inv tidy-all` in the repository, making some dependencies in the remote env outdated compared to your local env.
+- You didn't run `deva tidy` in the repository, making some dependencies in the remote env outdated compared to your local env.
```
Comment (chouetz) [inline docs/dev/linters.md:48]: Just a double check, you removed the `-all` part, if it was intentional it's ok

File: docs/public/setup.md [hunk 5]
```diff
@@ -33,10 +33,41 @@ On Windows, install Python 2.7 and/or 3.11 via the [official installer](https://
 
 ##### Preface
 
+[Invoke](http://www.pyinvoke.org) is a task runner written in Python that is extensively used in this project to orchestrate builds and test runs. To run the tasks, you need to have it installed on your machine. We offer two different ways to run our invoke tasks.
+
+##### `deva` (recommended)
```
Comment (chouetz) [inline docs/public/setup.md:38]: Could we factorise with what is on docs/dev/agent_dev_env.md?

**Review by chouetz — APPROVED** [review]
As it's "only" documentation feel free to handle or not my comments

**Review by chouetz — APPROVED** [review]
As it's "only" documentation feel free to handle or not my comments

File: docs/dev/linters.md (hunk 4 above)
Comment (FlorentClarret) [reply docs/dev/linters.md:48]: It is I deprecated it in favour of `tidy` 🙂

File: docs/dev/agent_dev_env.md (hunk 3 above)
Comment (FlorentClarret) [reply docs/dev/agent_dev_env.md:58]: added!

File: docs/public/setup.md (hunk 5 above)
Comment (FlorentClarret) [reply docs/public/setup.md:38]: I'm afraid we can't right now sadly

**Review by pgimalac — APPROVED** [review]

**Review by brett0000FF — APPROVED** [review]

**Pull request merged**